tests/e2e/matchers/to-have-valid-amp.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import amphtmlValidator from 'amphtml-validator';
+import jsdom from 'jsdom';
+const { JSDOM } = jsdom;
+/**
+ * WordPress dependencies
+ */
+import { createURL } from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import { fetchPageContent } from '../utils';
+
+/**
+ * Test for valid AMP for logged in users.
+ *
+ * @param {string} path The HTML to be used for matching elements.
+ *
+ * @return {Object} Object containing the pass flag and message for the matcher
+ */
+async function validateAMPforLoggedInUser( path ) {
+	let pass, message;
+	const urlToFetch = 'object' === typeof path ? path.url() : createURL( path );
+	const cookies = await page.cookies();
+
+	// Make sure we have a login cookie
+	expect( cookies.filter( ( cookie ) => cookie.name.match( /^wordpress_logged_in/ ) ).length ).toBeGreaterThan( 0 );
```

Comment:
This works, but would be a bit simpler to use `cookies.some()` rather than filtering through all of them as we're just looking for a single match.

I think this should work:
```suggestion
	expect( cookies.some( ( { name } ) => name.match( /^wordpress_logged_in/ ) ) ).toBeTruthy();
```

Reply:
updated